app/controllers/users.js

```diff
@@ -36,7 +36,40 @@ exports.login = (req, res, next) => {
   });
 };
 
-exports.update = (req, res, next) => {
+exports.invalidateAll = (req, res) => {
+  const user = req.user;
+  const SALT_ROUNDS = 10;
+
+  bcrypt.genSalt(SALT_ROUNDS).then((verificationCode) => {
+    user.verificationCode = verificationCode;
+    user.save();
+    res.status(200);
+    res.end();
+  }).catch(() => {
+    res.status(500);
+    res.end();
+  });
+};
+
+exports.renew = (req, res) => {
+  const accessToken = req.accessToken;
+  const user = req.user;
+
+  if (req.body.renew_id === accessToken.renewId) {
+    if (moment().isBefore(accessToken.expirationDateWarning)) {
+      res.status(403);
+      res.send({ error: 'Warning expiration date has not been reached' });
+    } else {
+      res.status(200);
+      res.send({ access_token: sessionManager.generateAccessTokenWithRenewId(user, accessToken.renewId) });
```

Comment:
the `generateAccesTokenWithRenewId` calculates each time the `maximumUsefulDate`, but here we shouldn't be updating that value, or am I missing something?